src/Middleware/RedirectToLogin.php

```diff
@@ -38,10 +38,31 @@ public function __construct(Horde_Registry $registry, ResponseFactoryInterface $
     }
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
+        
         if ($request->getAttribute('HORDE_AUTHENTICATED_USER')) {
             return $handler->handle($request);
         }
-        $redirect = (string)Horde::Url($this->registry->getInitialPage('horde'), true);
+
+        // set baseurl: check if alternative login is set
+        if(isset($GLOBALS['conf']['auth']['alternate_login'])){
+            $baseurl = $GLOBALS['conf']['auth']['alternate_login'];
+        }
+        else{
+            $baseurl = $this->registry->getInitialPage('horde');
+        };
+        
+        // create redirect url: first check if AppFinder found a legitimate app in url
```

Comment:
I think this can be removed. If it did not find an app, it would not even reach this middleware.

Reply:
ok, I thinkgs thats true